agent/acs/client/acs_client_test.go

```diff
@@ -169,8 +169,10 @@ func TestPayloadHandlerCalled(t *testing.T) {
 func TestRefreshCredentialsHandlerCalled(t *testing.T) {
 	cs, ml := testCS()
 
+	handlerCalled := make(chan struct{})
```

Comment:
Can you use a `sync.WaitGroup` instead? Just that it's intended to doing stuff like this and we might as well use instead of creating a channel ourselves.